### PR TITLE
fix(DetailsSection): fixes close section on body click

### LINKED ISSUE #120: Collapsible DetailsSection closes when body is clicked
A collapsible DetailsSection should only close when the header bar is clicked. User should be able to interact with the body without the section collapsing.

### PR DESCRIPTION
**Pull request by lucasrose**
closes #120

### CODE REVIEW

**Comment by adampickeral** [comment]
you shouldn't need to add `headers` to all those test cases. you just need to find the component with the `rs-detail-section-header` class and simulate a click on that

**Comment by adampickeral** [comment]
also, reference the issue that is being closed in the footer of your commit message

**Comment by akrikos** [comment]
:+1:

File: test/DetailsSectionSpec.jsx [hunk 1]
```diff
@@ -239,11 +245,25 @@ describe('DetailsSection', () => {
...
     expect(ReactDOM.findDOMNode(detailsSection)).toHaveClass('rs-detail-section rs-collapsible-section collapsed');
-    TestUtils.Simulate.click(ReactDOM.findDOMNode(detailsSection));
+    TestUtils.Simulate.click(ReactDOM.findDOMNode(header));
     expect(ReactDOM.findDOMNode(detailsSection)).toHaveClass('rs-detail-section rs-collapsible-section expanded');
-    TestUtils.Simulate.click(ReactDOM.findDOMNode(detailsSection));
+    TestUtils.Simulate.click(ReactDOM.findDOMNode(header));
     expect(ReactDOM.findDOMNode(detailsSection)).toHaveClass('rs-detail-section rs-collapsible-section collapsed');
   });
+
+  it('section doesnt expand when clicking on body', () => {
```
Comment (adampickeral) [inline test/DetailsSectionSpec.jsx]: doesnt collapse

File: test/DetailsSectionSpec.jsx [hunk 2]
```diff
@@ -198,38 +199,43 @@ describe('DetailsSection', () => {
...
+    TestUtils.Simulate.click(ReactDOM.findDOMNode(header));
     expect(currentState).toBe(false);
   });
 
   it('section initally collapsed updates currentState with onToggleCollapsed', () => {
     detailsSection = TestUtils.renderIntoDocument(
-      <DetailsSection title="title" collapsible={ true } initialCollapsed={ currentState = true } onToggleCollapsed={ toggleState }>
-        Test Detail List
+      <DetailsSection title="title" collapsible={ true } initialCollapsed={ currentState = true } onToggleCollapsed={ toggleState }
+       >
```
Comment (adampickeral) [inline test/DetailsSectionSpec.jsx]: :stuck_out_tongue_closed_eyes:

**Pull request merged**